### PR TITLE
feat(jobs): drain zombie NATS streams in periodic health check

### PR DESCRIPTION
**Pull request by mihow**
## What this does (plain English)

When an ML job finishes or gets cancelled, we sometimes leave its NATS message queue behind. Workers don't know the job is over, so they keep politely asking that queue for work forever — burning cycles on nothing. This PR adds a periodic cleanup that deletes those orphaned queues once they're clearly abandoned, while keeping a safety window so we don't accidentally tear down a queue that a just-started job is still setting up.

---

## Summary

Adds a `zombie_streams` sub-check to the existing `jobs_health_check` periodic beat task. Drains NATS JetStream `job_{N}` streams whose backing Django Job is in a terminal state (or missing) AND whose stream age exceeds `Job.ZOMBIE_STREAMS_MAX_AGE_MINUTES` (default 60 minutes).

Zombie streams accumulate when the cleanup-on-cancel path misses a stream on job failure or revocation; workers keep polling the leftover subject for no reason. This is defense-in-depth — the underlying cleanup-on-cancel gap is a separate fix.

Two guards protect against draining live state:

- **Terminal-only**: streams are only drained if the Job is in `JobState.final_states()` or the Job row is missing. In-flight Jobs are skipped.
- **Age guard**: streams newer than `ZOMBIE_STREAMS_MAX_AGE_MINUTES` are skipped. Protects against the `transaction.on_commit` race where a NATS stream exists briefly before the Job row is persisted.

Implementation notes:

- Uses the raw `$JS.API.STREAM.LIST` endpoint (in `ami/ml/orchestration/nats_queue.py::list_job_stream_snapshots`) because `JetStreamContext.streams_info` in the currently pinned `nats.py` drops the server-side `created` timestamp, which we need to age zombies out with a safety margin. Pages at 256 streams per response so long tails are still fully enumerated.
- Returns `{checked, fixed, unfixable}` counters consistent with the other sub-checks in `JobsHealthCheckResult`.

## Test plan

- [x] `ami.jobs.tests.test_periodic_beat_tasks` — unit tests cover drain, skip-fresh, skip-running, missing-job, paginated listing
- [x] E2E on local stack against a real NATS server: drain fires on terminal+old, skips fresh+terminal, skips old+running

## E2E validation

Invoked `jobs_health_check()` synchronously against a live local NATS server in three scenarios. All three hit the expected branch of the age/state guard.

**1. Drain on terminal+old.** Created stream `job_999991` with no matching `Job` row, patched `Job.ZOMBIE_STREAMS_MAX_AGE_MINUTES = 0`, ran the check:

```
zombie_streams: checked=1, fixed=1, unfixable=0
INFO Drained zombie NATS stream job_999991 (status=missing, age=0.0h, redelivered=None, consumer_deleted=False)
```

`consumer_deleted=False` is expected — the fixture only created a stream, no consumer. The stream itself (the entity that was wasting worker poll cycles) was cleanly removed. Post-drain `stream_info` raised `NotFoundError` as expected.

**2. Skip on fresh+terminal** (the `transaction.on_commit` race guard). Created a fresh stream `job_999992`, left the threshold at the default 60 min, ran the check:

```
zombie_streams: checked=1, fixed=0, unfixable=0
```

Stream retained. This is the scenario that would break async_api dispatch if it regressed — a freshly-created stream whose `Job` row hasn't committed yet must not be drained.

**3. Skip on old+running.** Created a throwaway `Job` (status=`STARTED`) and matching stream, patched threshold to 0 so age alone would permit drain:

```
zombie_streams: checked=1, fixed=0, unfixable=0
```

Stream retained. In-flight Job correctly shields the stream even when the age guard would otherwise permit it. `running_job_snapshots` sub-check simultaneously picked up the Job as expected (`checked=1`).

All three scenarios also exercise the raw `$JS.API.STREAM.LIST` enumeration + UTC→local timestamp conversion against the real NATS server, which is the piece the unit tests mock.

## Release Notes

* **New Features**
  * Added automatic cleanup of stale job streams in NATS for jobs in terminal states or no longer tracked in the system.
  * Enhanced health check monitoring to include zombie stream detection and removal.

* **Tests**
  * Added comprehensive test coverage for zombie stream detection and cleanup behavior, including edge cases and failure scenarios.

### CODE REVIEW

File: ami/ml/orchestration/nats_queue.py [hunk 1]
```diff
@@ -573,6 +596,79 @@ async def delete_stream(self, job_id: int) -> bool:
...
+        while True:
+            resp = await asyncio.wait_for(
+                self.nc.request("$JS.API.STREAM.LIST", json.dumps({"offset": offset}).encode()),
+                timeout=NATS_JETSTREAM_TIMEOUT,
+            )
+            payload = json.loads(resp.data)
+            streams = payload.get("streams") or []
+            if not streams:
+                break
+            for stream in streams:
```
[the remark this answers is not in the document]
Comment (mihow) [reply ami/ml/orchestration/nats_queue.py:638]: **Claude says:** Fixed in 1e24d512. `list_job_stream_snapshots` now checks `payload.get('error')` and raises `nats.errors.Error`. `_safe_run_sub_check` records `unfixable=1`. Test: `test_list_job_stream_snapshots_raises_on_nats_error_payload`.

File: ami/ml/orchestration/nats_queue.py [hunk 2]
```diff
@@ -573,6 +596,79 @@ async def delete_stream(self, job_id: int) -> bool:
...
+                        "messages": int(state.get("messages") or 0),
+                        "num_redelivered": await self._consumer_redelivered_count(job_id),
+                    }
+                )
+            total = int(payload.get("total") or 0)
+            offset += len(streams)
+            if offset >= total:
+                break
+        return snapshots
+
```
[the remark this answers is not in the document]
Comment (mihow) [reply ami/ml/orchestration/nats_queue.py]: **Claude says:** Fixed in 1e24d512. Two-pass approach: `list_job_stream_snapshots` returns `num_redelivered=None`; `TaskQueueManager.populate_redelivered_counts(candidates, concurrency=8)` fills them for drain candidates only. O(N) per tick → O(candidates).

File: ami/jobs/tasks.py [hunk 3]
```diff
@@ -635,6 +636,91 @@ async def _snapshot_all() -> None:
...
+            jobs_by_id = await sync_to_async(
+                lambda ids: {j.pk: j for j in Job.objects.filter(pk__in=ids).only("pk", "status")}
+            )(job_ids)
+
+            checked = len(snapshots)
+            drained = 0
+            errored = 0
+            for snap in snapshots:
+                created = snap["created"]
+                age = now - created if created else threshold + datetime.timedelta(minutes=1)
```
[the remark this answers is not in the document]
Comment (mihow) [reply ami/jobs/tasks.py]: **Claude says:** Fixed in 1e24d512 — this was the most important one. Missing/unparseable `created` now *skips* the stream (WARNING, `unfixable++`) rather than being treated as older-than-threshold. Preserves the age guard's safety intent: unknown age ⇒ unsafe to drain. Test: `test_zombie_stream_skipped_when_created_is_none`.
